test/simulator/moves/ragepowder.js

```diff
@@ -27,15 +28,15 @@ describe('Rage Powder', function () {
 		let hitCount = [0, 0, 0];
 		battle.p1.active[0].damage = function (...args) {
 			hitCount[0]++;
-			return BattleEngine.BattlePokemon.prototype.damage.apply(this, args);
+			return BattlePokemon.prototype.damage.apply(this, args);
```

Comment:
The trick is that you don't need to know that ``this`` is ``BattlePokemon``.
This is just monkeypatching (or duck-punching as you call it, funny name I must say)

```js
instance.method = function () {
onMethodObserver();
Object.getPrototypeOf(this).method();
};
```

Too bad that ``super`` wasn't allowed there in the final ES6 spec :(